libs/agno/agno/knowledge/markdown.py

```diff
@@ -0,0 +1,141 @@
+from pathlib import Path
+from typing import Any, AsyncIterator, Dict, Iterator, List, Optional, Union
+
+from agno.document import Document
+from agno.document.reader.markdown_reader import MarkdownReader
+from agno.knowledge.agent import AgentKnowledge
+from agno.utils.log import log_info, logger
+
+
+class MarkdownKnowledgeBase(AgentKnowledge):
+    path: Optional[Union[str, Path, List[Dict[str, Union[str, Dict[str, Any]]]]]] = None
+    formats: List[str] = [".md"]
+    reader: MarkdownReader = MarkdownReader()
+
+    @property
+    def document_lists(self) -> Iterator[List[Document]]:
+        """Iterate over text files and yield lists of documents."""
+        if self.path is None:
+            raise ValueError("Path is not set")
+
+        if isinstance(self.path, list):
+            for item in self.path:
+                if isinstance(item, dict) and "path" in item:
```

Comment:
What if it is just a list of paths?

Reply:
Without metadata? This should also be allowed

Reply:
Yup, this works with a list of paths, we still yield the documents here if they don't have metadata. Also notice this logic is the same we have in `TextKnowledgeBase`